server/dns-service/dnsservice-library/src/main/java/org/apache/james/dnsservice/library/netmatcher/NetMatcher.java

```diff
@@ -18,164 +18,105 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-/**
- * NetMatcher Class is used to check if an ipAddress match a network.
- * 
- * NetMatcher provides a means for checking whether a particular IPv4 or IPv6
- * address or domain name is within a set of subnets.
- */
 public class NetMatcher {
-    public static final String NETS_SEPARATOR = ", ";
+    private static final String NETS_SEPARATOR = ", ";
+    private static final Logger LOG = LoggerFactory.getLogger(NetMatcher.class);
+    private static final Comparator<InetNetwork> INET_NETWORK_COMPARATOR = new Comparator<InetNetwork>() {
+        public int compare(InetNetwork in1, InetNetwork in2) {
+            return in1.toString().compareTo(in2.toString());
+        }
+    };
+    private static final Predicate<Optional<InetNetwork>> IS_PRESENT_INET_NETWORK = new Predicate<Optional<InetNetwork>>() {
+        @Override
+        public boolean apply(Optional<InetNetwork> inetNetworkOptional) {
+            return inetNetworkOptional.isPresent();
+        }
+    };
+    private static final Function<Optional<InetNetwork>, InetNetwork> GET_INET_NETWORK = new Function<Optional<InetNetwork>, InetNetwork>() {
+        @Override
+        public InetNetwork apply(Optional<InetNetwork> inetNetworkOptional) {
+            return inetNetworkOptional.get();
+        }
+    };
 
-    /**
-     * The DNS Service used to build InetNetworks.
-     */
     private final DNSService dnsServer;
+    private final SortedSet<InetNetwork> networks;
+    private final InetNetworkBuilder inetNetwork;
+    private final Collection<String> nets;
 
-    /**
-     * The Set of InetNetwork to match against.
-     */
-    private SortedSet<InetNetwork> networks;
-
-    /**
-     * Create a new instance of Netmatcher.
-     * 
-     * @param nets
-     *            a String[] which holds all networks
-     * @param dnsServer
-     *            the DNSService which will be used in this class
-     */
-    public NetMatcher(String[] nets, DNSService dnsServer) {
-        this.dnsServer = dnsServer;
-        initInetNetworks(nets);
-    }
-
-    /**
-     * Create a new instance of Netmatcher.
-     * 
-     * @param nets
-     *            a Collection which holds all networks
-     * @param dnsServer
-     *            the DNSService which will be used in this class
-     */
     public NetMatcher(Collection<String> nets, DNSService dnsServer) {
+        this.nets = nets;
         this.dnsServer = dnsServer;
-        initInetNetworks(nets);
+        this.inetNetwork = new InetNetworkBuilder(this.dnsServer);
+        this.networks = initInetNetworks();
     }
 
     public NetMatcher(String commaSeparatedNets, DNSService dnsServer) {
-        this.dnsServer = dnsServer;
-        List<String> nets = Splitter.on(NETS_SEPARATOR).splitToList(commaSeparatedNets);
-        initInetNetworks(nets);
+        this(Splitter.on(NETS_SEPARATOR).splitToList(commaSeparatedNets), dnsServer);
```

Comment:
Sorry but I don't understand what do you mean?
Do you mean create new class NetMatcherFactory which create instance of NetMatcher? I tried with specified commit, could you take a look at? Thank you.